src/averageAges.js

```diff
@@ -38,6 +54,18 @@ function calculateMenAverageAge(people, century) {
  */
 function calculateWomenAverageAge(people, withChildren) {
   // write code here
+  const WEMEN = withChildren
+    ? people
+      .filter((person, i, persons) => person.sex === 'f'
+        && persons
+          .some(p => p.mother === person.name))
+    : people.filter(person => person.sex === 'f');
+
+  const WEMEN_AGES = WEMEN
+    .reduce((total, ages) => total + ages.died - ages.born, 0);
+  const AVARAGE_AGE = WEMEN_AGES / WEMEN.length.toFixed(2);
+
+  return AVARAGE_AGE;
```

Comment:
```suggestion
  return calculateAverageAge(women);
```
You're doing the same calculations twice at this point, consider extracting that logic to a separate function.